nav2_lifecycle_manager/src/lifecycle_manager.cpp

```diff
@@ -248,9 +256,95 @@ LifecycleManager::resume()
   return true;
 }
 
-// TODO(mjeronimo): This is used to emphasize the major events during system bring-up and
-// shutdown so that the messgaes can be easily seen among the log output. We should replace
-// this with a ROS2-supported way of highlighting console output, if possible.
+void
+LifecycleManager::createBondTimer()
+{
+  if (bond_timeout_.count() <= 0) {
+    return;
+  }
+
+  message("Creating bond timer...");
+
+  bond_timer_ = this->create_wall_timer(
+    100ms,
+    std::bind(&LifecycleManager::checkBondConnections, this));
+}
+
+void
+LifecycleManager::destroyBondConnections()
+{
+  bond_timer_.reset();
+
+  if (bond_map_.empty()) {
+    return;
+  }
+
+  message("Terminating bond connections...");
+
+  for (auto & node_name : node_names_) {
+    bond_map_[node_name]->breakBond();
+  }
+
+  bond_map_.clear();
+}
+
+void
+LifecycleManager::createBondConnections()
+{
+  const double timeout_ns =
+    std::chrono::duration_cast<std::chrono::nanoseconds>(bond_timeout_).count();
+  const double timeout_s = timeout_ns / 1e9;
+
+  for (auto & node_name : node_names_) {
+    bond_map_[node_name] =
+      std::make_shared<bond::Bond>("bond", node_name, service_client_node_);
+    auto & node_bond = bond_map_[node_name];
+    node_bond->setHeartbeatTimeout(timeout_s);
+    node_bond->start();
+
+    RCLCPP_INFO(get_logger(), "Server %s trying to connect to bond....", node_name.c_str());
+    node_bond->waitUntilFormed(rclcpp::Duration(timeout_ns));
+    if (node_bond->isBroken()) {
+      RCLCPP_ERROR(
+        get_logger(),
+        "Server %s was unable to be reached after %0.2fs by bond. "
+        "This server may be misconfigured.",
+        node_name.c_str(), timeout_s, node_name.c_str());
```

Comment:
Is last the `node_name.c_str()` extra?

Reply:
Yes, that was from an older warning, thanks